packages/client/lib/AccessTokenClient.ts

```diff
@@ -130,16 +136,26 @@ export class AccessTokenClient {
     }
   }
 
-  private isPinRequiredValue(requestPayload: CommonCredentialOfferRequestPayload): boolean {
+  private isPinRequiredValue(requestPayload: CredentialOfferPayload): boolean {
     let isPinRequired = false;
-    if (requestPayload !== undefined) {
+    if (!requestPayload) {
+      throw new Error(TokenErrorResponse.invalid_request);
+    }
+    const issuer = getIssuerFromCredentialOfferPayload(requestPayload);
+    if (isCredentialOfferV1_0_09(requestPayload)) {
+      requestPayload = requestPayload as CredentialOfferV1_0_09;
       if (typeof requestPayload.user_pin_required === 'string') {
         isPinRequired = requestPayload.user_pin_required.toLowerCase() === 'true';
       } else if (typeof requestPayload.user_pin_required === 'boolean') {
         isPinRequired = requestPayload.user_pin_required;
       }
+    } else if (isCredentialOfferV1_0_11(requestPayload)) {
+      requestPayload = requestPayload as CredentialOfferPayloadV1_0_11;
+      if ('grants' in requestPayload && 'urn:ietf:params:oauth:grant-type:pre-authorized_code' in requestPayload.grants!) {
+        isPinRequired = requestPayload!.grants!['urn:ietf:params:oauth:grant-type:pre-authorized_code']!.user_pin_required;
```

Comment:
// eslint-disable-next-line @typescript-eslint/no-non-null-assertion

Reply:
added the above comment